cmd/thanos/compact.go

```diff
@@ -326,14 +352,25 @@ func runCompact(
 			level.Warn(logger).Log("msg", "downsampling was explicitly disabled")
 		}
 
-		if err := compact.ApplyRetentionPolicyByResolution(ctx, logger, bkt, metaFetcher, retentionByResolution); err != nil {
+		if err := compact.ApplyRetentionPolicyByResolution(ctx, logger, bkt, metaFetcher, retentionByResolution, blocksMarkedForDeletion); err != nil {
```

Comment:
I think here we need to pass Syncer's `prometheus.Counter`. Because otherwise we will have two metrics now:
* `thanos_compactor_blocks_marked_for_deletion` from this
* `thanos_compact_blocks_marked_for_deletion_total` from the Syncer
We should probably have one unified metric. Also, one of them doesn't follow the Prometheus conventions :p

Reply:
Probably `thanos_compactor_blocks_marked_for_deletion`  and `thanos_compact_syncer_blocks_marked_for_deletion_total` will suffice ?
Or do you think passing metric in constructor is better option?

Reply:
@GiedriusS is right, we need one. And for counters it's always `_total` suffix required. Let's either pass to syncer that counter or use the one from syncer. Definitely we need one counter (:

Reply:
Since we require only one counter, I will pass this counter as an argument to syncer, that way we track it through one metric only.

Reply:
I am resolving this thread since I am using one metric now. Please open a new thread if required.